File: javascript/1094.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n')
var testes = parseInt(lines.shift())

var [coelhos,sapos,ratos,total] = [0,0,0,0]

for(let i=0;i<=testes-1;i++){
    var experimento = lines[i].split(' ')
    var currentQuant = parseInt(experimento[0])
    var currentType = String(experimento[1].trim())
    total+=currentQuant
    switch(currentType){
        case 'R':
            ratos+=currentQuant
            break;
        case 'S':
            sapos+=currentQuant
            break;
        case 'C':
            coelhos+=currentQuant
    }
}
var percCoelhos = (100 * coelhos)/total
var percRatos = (100 * ratos)/total
var percSapos = (100 * sapos)/total
console.log(`Total: ${total} cobaias`);
console.log(`Total de coelhos: ${coelhos}`);
console.log(`Total de ratos: ${ratos}`);
console.log(`Total de sapos: ${sapos}`);
console.log(`Percentual de coelhos: ${percCoelhos.toFixed(2)} %`);
console.log(`Percentual de ratos: ${percRatos.toFixed(2)} %`);
console.log(`Percentual de sapos: ${percSapos.toFixed(2)} %`);

File: python/1094.py
testes=int(input())
total=0
coelhos=0
ratos=0
sapos=0

for x in range(1,testes+1):
    currentTest=input().split(' ')
    currentQuant=int(currentTest[0])
    currentType=str(currentTest[1])
    
    total=total+currentQuant
    if currentType == 'R':
        ratos=ratos+currentQuant
    elif currentType == 'S':
        sapos=sapos+currentQuant
    else:
        coelhos=coelhos+currentQuant
    
percCoelho= '{:.2f}'.format((100 * coelhos)/total)
percRato= '{:.2f}'.format((100 * ratos)/total)
percSapo= '{:.2f}'.format((100 * sapos)/total)

print('Total: '+str(total)+' cobaias')
print('Total de coelhos: '+str(coelhos))
print('Total de ratos: '+str(ratos))
print('Total de sapos: '+str(sapos))
print('Percentual de coelhos: '+str(percCoelho)+' %')
print('Percentual de ratos: '+str(percRato)+' %')
print('Percentual de sapos: '+str(percSapo)+' %')